common/helpers/frameworks/render-nomis-mappings-to-field.test.js

```diff
@@ -0,0 +1,125 @@
+const proxyquire = require('proxyquire')
+
+const i18n = require('../../../config/i18n')
+const frameworkNomisMappingsToPanelStub = sinon.stub().returns('STUB_MAPPINGS')
+const helper = proxyquire('./render-nomis-mappings-to-field', {
+  '../../presenters/framework-nomis-mappings-to-panel': frameworkNomisMappingsToPanelStub,
+})
+
+describe('#renderNomisMappingsToField', function () {
+  let output
+  const mockField = [
+    'mock-key',
+    {
+      foo: 'bar',
+    },
+  ]
+
+  beforeEach(function () {
+    sinon.stub(i18n, 't').returnsArg(0)
+  })
+
+  context('without response', function () {
+    beforeEach(function () {
+      output = helper()(mockField)
+    })
+
+    it('should return original field', function () {
+      expect(output).to.deep.equal(mockField)
+    })
+  })
+
+  context('with response', function () {
+    const mockResponse = {
+      question: {
+        key: 'mock-key',
+      },
+    }
+
+    context('without NOMIS mappings', function () {
+      beforeEach(function () {
+        output = helper([mockResponse])(mockField)
+      })
+
+      it('should return original field', function () {
+        expect(output).to.deep.equal(mockField)
+      })
+    })
+
+    context('with NOMIS mappings', function () {
+      const mockResponseWithMappings = {
```

Comment:
I've had weird runtime issues in the past with setting variables inside a `describe/context` block but outside of a `before/beforeEach/it` block, so now I declare a variable outside, and use a `before/beforeEach` to set it - even if they're not going to change.

Reply:
Hmm, that's interesting. I've not experienced that before with defining strings/objects but have had that when defining stubs that might need resetting. 

I think we also use this approach in quite a few places within our test suites. Have you had any of those issues running the tests in this project yet? If it's cropped up it might be good to look into to see why it's happening.

I'm happy to change this up, but also wary that we do use this approach in quite a few other places so if it's something we should be doing, maybe we should do it as a global change and then we can also explain the reasons behind it alongside the change?

Reply:
No worries about doing it this change, I guess its now more of an fyi.

What happens is that mocha runs all the code outside of the before blocks while starting up the tests, which can cause other unrelated tests to fail. It's an edge case, but Ive seen it in several projects - haven't seen it causing issues here yet.

Happy to ignore it for now.